src/shim/runtime/v2/rune/v2/create.go

```diff
@@ -0,0 +1,140 @@
+	if err != nil {
+		return nil, err
+	}
+	sandboxNamspace, err := oci.SandboxNamespace(*ociSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	var container *runc.Container
+
+	switch containerType {
+	case shimtypes.PodSandbox:
+		container, err = runc.NewContainer(ctx, s.platform, r)
+		if err != nil {
+			return nil, err
+		}
+
+		if sandboxNamspace != shimtypes.KubeSystem {
+			ar := &taskAPI.CreateTaskRequest{
+				ID:       generateID(),
+				Terminal: false,
+				Options:  r.Options,
+			}
+
+			// Create agent enclave container
+			agentContainer, err := createAgentContainer(ctx, s, ar)
+			if err != nil {
+				return nil, err
+			}
+
+			if err := writeAgentIdFile(r.Bundle, ar.ID); err != nil {
+				return nil, err
+			}
+
+			s.agentID = ar.ID
+			s.pauseID = r.ID
+			s.containers[ar.ID] = agentContainer
+			s.agent = &agent{
+				ID:     agentContainer.ID,
+				Bundle: agentContainer.Bundle,
+				URL:    AgentUrl,
+			}
+		}
+	case shimtypes.PodContainer:
+		if sandboxNamspace != shimtypes.KubeSystem {
+			image, err := oci.GetImage(*ociSpec)
+			if err != nil {
+				return nil, err
+			}
+			cid, err := getContainerID(image)
+			if err != nil {
+				return nil, err
+			}
+
+			// Create upperDir and workDir for app container
+			upperDir := filepath.Join(ContainerBase, r.ID, "upper")
+			workDir := filepath.Join(ContainerBase, r.ID, "work")
+			for _, dir := range []string{upperDir, workDir} {
+				if err := os.MkdirAll(dir, 0711); err != nil && !os.IsExist(err) {
+					return nil, err
+				}
+			}
+			// sefsDir store the unionfs images (based on sefs)
+			sefsDir := filepath.Join(agentContainerRootDir, s.agentID, "merged/rootfs/images", cid)
+
+			var options []string
+			// Set index=off when mount overlayfs
+			options = append(options, "index=off")
+			options = append(options,
+				fmt.Sprintf("workdir=%s", workDir),
+				fmt.Sprintf("upperdir=%s", upperDir),
+			)
+			options = append(options, fmt.Sprintf("lowerdir=%s:%s", sefsDir, filepath.Join(bootContainerPath, "rootfs")))
+			r.Rootfs = append(r.Rootfs, &types.Mount{
+				Type:    "overlay",
+				Source:  "overlay",
+				Options: options,
+			})
+
+			// Update the Root.Path field in container spec from
+			// "/var/lib/containerd/io.containerd.grpc.v1.cri/containers/<id>/rootfs" to "rootfs"
+			// TODO: config.json will be updated by agent enclave container
+			err = config.UpdateRootPathConfig(filepath.Join(r.Bundle, "config.json"), "rootfs")
+			if err != nil {
+				return nil, err
+			}
+
+			shimLog.WithFields(logrus.Fields{
+				"Rootfs": r.Rootfs,
+				"Bundle": r.Bundle,
+			}).Debug("Create app enclave container based on sefs image")
+		}
+
+		container, err = runc.NewContainer(ctx, s.platform, r)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return container, nil
+}
+
+// ReadAgentIdFile reads the agent container id information from the path
+func readAgentIdFile(path string) (string, error) {
+	data, err := os.ReadFile(filepath.Join(path, "agent.id"))
```

Comment:
- What create this file?
- ooi, why is it not named something like `agent-id`?

Reply:
When the agent enclave container is created (RunSanbox 
stage), the container id of agent enclave container id will be written in `agent-id` file (writeAgentIdFile function). 
According to the container in the `agent-id` file, we can get the resource of the agent enclave container, then clean up the resource of the agent container when stopping the pod sandbox.